src/lib.rs

```diff
@@ -319,10 +321,13 @@ fn do_write(mut remaining: &[u8]) {
             grant[offset..][..min].copy_from_slice(&remaining[..min]);
             offset += min;
 
+            BBQ_GRANT_W.offset.store(offset, Ordering::Relaxed);
```

Comment:
So, this line is currently handled here: https://github.com/jamesmunns/defmt-bbq/blob/abb45806a8015d0280278057e32453c25430cb5c/src/lib.rs#L332

It's only necessary to store the offset if we *don't* commit the grant, as the offset only tracks the offset in the *current* grant. If we commit it, it goes back to zero.